File: 0920-uncommon-words-from-two-sentences/0920-uncommon-words-from-two-sentences.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
         # Combine both sentences
        s = s1 + " " + s2

        # Dictionary to store the word counts
        word_count = defaultdict(int)

        # Temporary variable to store each word
        tmp = ""
        for val in s:  # Iterate through each character in the combined string
            if val == ' ':
                word_count[tmp] += 1
                tmp = ""
            else:
                tmp += val
        
        # If there's any word left in `tmp`, add it to the word count
        if tmp != "":
            word_count[tmp] += 1

        # List to store the uncommon words (words with count == 1)
        ans = [word for word, count in word_count.items() if count == 1]

        return ans

File: 0920-uncommon-words-from-two-sentences/0920-uncommon-words-from-two-sentences.go
func uncommonFromSentences(s1 string, s2 string) []string {
    // Combine both sentences
	s := s1 + " " + s2

	// Create a map to store word counts
	wordCount := make(map[string]int)

	// Split the combined string by spaces into words
	words := strings.Fields(s)

	// Count occurrences of each word
	for _, word := range words {
		wordCount[word]++
	}

	// List to store uncommon words (words with count == 1)
	var ans []string
	for word, count := range wordCount {
		if count == 1 {
			ans = append(ans, word)
		}
	}

	return ans
}